### ISSUE TITLE
Instability calculation performance issues

### ISSUE THREAD

**Issue by sfPlayer1** [XCompWiz/Mystcraft-Issues#43]
I've noticed 2 issues with the instability calculation:
- It runs off both phases of the server tick (start+end), thus twice per tick, while probably only once is intended.
- com.xcompwiz.mystcraft.world.ChunkProviderMyst.mapVanillaToLocal() and com.xcompwiz.mystcraft.world.ChunkProviderMyst.mapLocalToVanilla() use comparatively slow reflection based array manipulation. It's so slow that Array.get and Array.set use 2/3rd of the server tick time, much more than the actual world generation itself. I recommend to just duplicate the 2 map methods and to use the explicit types directly.

**Comment by sfPlayer1** [comment]
Another issue: WorldProviderMystDummy.ChunkProviderServerDummy accumulates way too many chunks, 8574 in a heap dump I'm checking right now. This causes servers to run out of memory if they don't have much, like the typical medium range game servers offers with around 2 GB.

**Comment by XCompWiz** [comment]
These are good points. Thanks for finding them.
The double run per tick is indeed a bug. I'll fix that this very moment.

I hadn't paid much attention to the reflection on those mappings. I'll make it less general and more concrete, and that should improve them.
Thank you very much for the profiling information. :)

That's odd, as it's not supposed to keep more than 32 or so chunks, I think.  Certainly not so many as you see there.  I'll look into it.

**Comment by XCompWiz** [comment]
First two of these are knocked out. No issues.

I'm looking into the latter one. Looks like the removal from the list may not be working properly.
...
Yup, the indexing is completely wrong; not using the z coord at all, just the x coord twice.  This should fix it. We'll see.

**Comment by XCompWiz** [comment]
Yup. The chunks are reduced to 32 every unload cycle, FIFO style. All good now.
The world unloads properly, it seems, and the remaining 32 chunks should go with it.

Thanks again. These fixes will all be in the next build. :)
